include/aspect/newton.h

```diff
@@ -0,0 +1,116 @@
+/*
+  Copyright (C) 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#ifndef __aspect__newton_h
+#define __aspect__newton_h
+
+#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+#include <aspect/assembly.h>
+#include <aspect/material_model/interface.h>
+
+#include <deal.II/base/signaling_nan.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  namespace MaterialModel
+  {
+
+    template <int dim>
+    class MaterialModelDerivatives : public AdditionalMaterialOutputs<dim>
```

Comment:
Are these going to be all derivatives in the future because right now they are only viscosity derivatives for the Stokes block. If this stays like this, it should be reflected in the name.